### PR TITLE
Pylint fixes for three modules

### PR DESCRIPTION
**Pull request by asankov**
I had already made the changes in `tcms.report.views` when I realised that @ivo0126 has already fixes some issues in this module in #428 . However, I see that most of the changes are unrelated and non-duplicated. Yet, it is better to wait, until his PR is merged. Afterwards, I will rebase on top of him and fix the conflicts.

### CODE REVIEW

**Comment by atodorov** [comment]
Drop the commit related to reports (or move it to another branch so you don't lose it) and I can review this PR.

**Comment by asankov** [comment]
Done. This commit is now in a separate branch, ready to be added when the time has come.

File: tcms/testplans/helpers/email.py [hunk 1]
```diff
@@ -31,4 +31,4 @@ def get_plan_notification_recipients(plan):
 
     # don't email author of last change
     recipients.discard(getattr(plan.history.latest().history_user, 'email', ''))
-    return list(filter(lambda e: bool(e), recipients))
+    return list(filter(lambda e: bool, recipients))
```
Comment (atodorov) [inline tcms/testplans/helpers/email.py]: This is wrong, I think you should remove the entire first parameter of `filter()`. However you should write a small unit test for this method to make sure it doesn't change behavior.

Comment (atodorov) [reply tcms/testplans/helpers/email.py]: Drop this commit, already fixed by @ivo0126. The rest of this PR should be good to go.

File: tcms/search/advanced_search.py [hunk 2]
```diff
@@ -63,9 +63,9 @@ def _render_errors(request, data, forms):  # pylint: disable=inconsistent-return
         product_choice = []
         for product in Product.objects.all():
             product_choice.append((product.pk, product.name))
-        plan_type_choices = PlanType.objects.all()  # pylint: disable=possibly-unused-variable
+        plan_type_choices = PlanType.objects.all()  # pylint: disable=unused-variable
         errors = _fmt_errors(errors)
-        priorities = Priority.objects.filter(  # pylint: disable=possibly-unused-variable
```
Comment (atodorov) [inline tcms/search/advanced_search.py]: This is actually the correct spelling. The fix from this PR causes the errors to appear again, see the travis log.

**Pull request closed without merging**